core/services/ocr2/delegate.go

```diff
@@ -290,7 +290,7 @@ func (d *Delegate) cleanupEVM(jb job.Job, q pg.Queryer, relayID relay.ID) error
 
 	rargs := types.RelayArgs{
 		ExternalJobID: jb.ExternalJobID,
-		JobID:         spec.ID,
+		JobID:         jb.ID,
```

Comment:
can we change on of the tests that should have caught this problem so it can't come back?

Reply:
https://smartcontract-it.atlassian.net/browse/BCF-2666